docs/pages/setup/reference/cli.mdx

```diff
@@ -688,6 +688,21 @@ Environment variables configure your tsh client and can help you avoid using fla
 | TELEPORT_ADD_KEYS_TO_AGENT | Specifies if the user certificate should be stored on the running SSH agent | yes, no, auto, only |
 | TELEPORT_USE_LOCAL_SSH_AGENT | Disable or enable local SSH agent integration | true, false |
 
+### tsh configuration file
+
+`tsh` has an optional configuration file that is stored in `$HOME/.tsh/config/config.yml`.
+
+The `tsh` configuration file enables you to specify HTTP headers to be
+included in requests to Teleport Proxy Servers with addresses matching
+the `proxy` field.
+
+```yaml
+add_headers:
+  - proxy: "*.example.com" # matching proxies will have headers included
+    headers: # headers are pairs to include in the http headers
+      foo: bar # Key/Value to be included in the http request
+```
+
```

Comment:
I think it would be great to include an example of when you would use custom HTTP headers in Proxy requests. For example, we could talk about the use case of sending traffic through intermediary proxies that require opaque auth tokens in HTTP headers (https://github.com/gravitational/teleport/pull/9475/files).

Otherwise, users may not know whether this config file is useful for their setup.

Also, are there any values of `headers` that users should avoid?

Reply:
> Also, are there any values of headers that users should avoid?

I'm not aware of any, there doesnt seem to be anything else modifying headers in the webclient code either, so I think all values should be okay.